api-admin/src/main/java/bio/terra/pearl/api/admin/service/MailingListExtService.java

```diff
@@ -32,4 +46,30 @@ public List<MailingListContact> getAll(
         portalEnvironmentService.findOne(portal.getShortcode(), envName).get();
     return mailingListContactService.findByPortalEnv(portalEnv.getId());
   }
+
+  @Transactional
+  public void delete(
+      String portalShortcode, EnvironmentName envName, UUID contactId, AdminUser user) {
+    Portal portal = authUtilService.authUserToPortal(user, portalShortcode);
+    PortalEnvironment portalEnv =
+        portalEnvironmentService.findOne(portal.getShortcode(), envName).get();
+    MailingListContact contact = mailingListContactService.find(contactId).get();
+    if (!contact.getPortalEnvironmentId().equals(portalEnv.getId())) {
+      throw new PermissionDeniedException("Contact does not belong to the given portal");
+    }
+    try {
+      DataChangeRecord changeRecord =
+          DataChangeRecord.builder()
+              .modelName(MailingListContact.class.getSimpleName())
+              .responsibleAdminUserId(user.getId())
+              .portalEnvironmentId(portalEnv.getId())
+              .newValue(null)
+              .oldValue(objectMapper.writeValueAsString(contact))
+              .build();
+      dataChangeRecordService.create(changeRecord);
+    } catch (JsonProcessingException e) {
+      throw new RuntimeException("could not create audit trail", e);
+    }
```

Comment:
I expected to see this in `MailingListContactService`. We probably don't want to `delete` without writing a change record, so we may as well limit the possibility of that happening.

Reply:
I thought about that -- the trouble is that we sometimes DO need to delete without audit trail, like when we're doing populates.  And right now the expectations of the core services is "I just do whatever CRUD operations you tell me to".    So I put it here for now, you're right that this isn't yet the optimal configuration.